pkg/controller/container-runtime-config/container_runtime_config_controller.go

```diff
@@ -523,71 +517,79 @@ func (ctrl *Controller) syncContainerRuntimeConfig(key string) error {
 		if err != nil {
 			return ctrl.syncStatusOnly(cfg, err)
 		}
-		if err := retry.RetryOnConflict(updateBackoff, func() error {
-			mc, err := ctrl.client.MachineconfigurationV1().MachineConfigs().Get(context.TODO(), managedKey, metav1.GetOptions{})
-			if err != nil && !errors.IsNotFound(err) {
-				return ctrl.syncStatusOnly(cfg, err, "could not find MachineConfig: %v", managedKey)
-			}
-			isNotFound := errors.IsNotFound(err)
-			// Generate the original ContainerRuntimeConfig
-			originalStorageIgn, _, _, err := generateOriginalContainerRuntimeConfigs(ctrl.templatesDir, controllerConfig, role)
-			if err != nil {
-				return ctrl.syncStatusOnly(cfg, err, "could not generate origin ContainerRuntime Configs: %v", err)
-			}
-
-			var configFileList []generatedConfigFile
-			ctrcfg := cfg.Spec.ContainerRuntimeConfig
-			if ctrcfg.OverlaySize != (resource.Quantity{}) {
-				storageTOML, err := ctrl.mergeConfigChanges(originalStorageIgn, cfg, updateStorageConfig)
-				if err != nil {
-					glog.V(2).Infoln(cfg, err, "error merging user changes to storage.conf: %v", err)
-				} else {
-					configFileList = append(configFileList, generatedConfigFile{filePath: storageConfigPath, data: storageTOML})
-				}
+		mc, err := ctrl.client.MachineconfigurationV1().MachineConfigs().Get(context.TODO(), managedKey, metav1.GetOptions{})
+		if err != nil && !errors.IsNotFound(err) {
+			return ctrl.syncStatusOnly(cfg, err, "could not find MachineConfig: %v", managedKey)
+		}
+		isNotFound := errors.IsNotFound(err)
+		// If we have seen this generation and the sync didn't fail, then skip
+		if !isNotFound && cfg.Status.ObservedGeneration >= cfg.Generation && cfg.Status.Conditions[len(cfg.Status.Conditions)-1].Type == mcfgv1.ContainerRuntimeConfigSuccess {
+			// But we still need to compare the generated controller version because during an upgrade we need a new one
+			mcCtrlVersion := mc.Annotations[ctrlcommon.GeneratedByControllerVersionAnnotationKey]
```

Comment:
So this check is the crux of the change for the bug right? To still sync even if we've observed a generation?